### ISSUE TITLE
upload to folder with dash in name creates wrong path

### ISSUE THREAD

**Issue by gradel** [stephenmcd/filebrowser-safe#60]
16b4dd714d65a0ccfcce4d436e30342a2d6b3a77 introduced this bug:

trying to upload file to "test-minus" folder creates "test3\u002Dminus" and fails with:
```
Internal Server Error: /de/admin/media-library/upload_file/
Traceback (most recent call last):
  File "/home/gerald/projects/mittelhof/mittelhof/env/lib/python3.4/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/gerald/projects/mittelhof/mittelhof/env/lib/python3.4/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3.4/contextlib.py", line 30, in inner
    return func(*args, **kwds)
  File "/home/gerald/projects/mittelhof/mittelhof/env/lib/python3.4/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/gerald/projects/mittelhof/mittelhof/env/src/filebrowser-safe/filebrowser_safe/decorators.py", line 25, in decorator
    return function(request, *args, **kwargs)
  File "/home/gerald/projects/mittelhof/mittelhof/env/lib/python3.4/site-packages/django/contrib/auth/decorators.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/home/gerald/projects/mittelhof/mittelhof/env/src/filebrowser-safe/filebrowser_safe/views.py", line 343, in _upload_file
    default_storage.move(smart_text(uploadedfile), smart_text(file_path), allow_overwrite=True)
  File "/home/gerald/projects/mittelhof/mittelhof/env/src/filebrowser-safe/filebrowser_safe/storage.py", line 60, in move
    file_move_safe(self.path(old_file_name), self.path(new_file_name), allow_overwrite=True)
  File "/home/gerald/projects/mittelhof/mittelhof/env/lib/python3.4/site-packages/django/core/files/move.py", line 57, in file_move_safe
    with open(old_file_name, 'rb') as old_file:
FileNotFoundError: [Errno 2] No such file or directory: '/home/gerald/projects/mittelhof/mittelhof/static/media/uploads/test3/u002Dminus/4136-900x585.jpg'
```

![fb-admin](https://cloud.githubusercontent.com/assets/635870/12871494/54be10cc-cd7e-11e5-9634-985cf41d6b56.png)

python 3.4, django 1.9.1, mezzanine master

**Comment by stephenmcd** [comment]
That's not an ascii hyphen, it was probably copied from MS word or similar.

**Comment by gradel** [comment]
well, created the folder in filebrowser admin itself.

**Comment by gradel** [comment]
and occurs only if the hyphened folder is in the top level O_o

**Comment by gradel** [comment]
Please have a look here:

![fb-dir-bug](https://cloud.githubusercontent.com/assets/635870/12914176/22d2e6d6-cf23-11e5-899f-7d235e4186e6.png)

removing the "escapejs" templatefilter from the "folder" input field in
filebrowser_safe/templates/filebrowser/upload.html
resolves this.
Time to reopen?

**Comment by stephenmcd** [comment]
Sorry, I misinterpreted the error originally. I've applied your fix in bf074a4f04fe371bdb33c1ef4908e4ddb9987f55

Thanks a lot.
